src/include/processor/operator/persistent/reader/csv/driver.h

```diff
@@ -58,17 +60,20 @@ class SerialParsingDriver : public ParsingDriver {
     BaseCSVReader* getReader() override;
 };
 
-struct SniffCSVNameAndTypeDriver {
-    std::vector<std::pair<std::string, common::LogicalType>> columns;
-    std::vector<bool> sniffType;
-    // if the type isn't declared in the header, sniff it
+class SniffCSVNameAndTypeDriver : public SerialParsingDriver {
     SerialCSVReader* reader;
```

Comment:
Isn't this shadowing the `reader` of `SerialParsingDriver` at L50? I don't think this is properly set during construction.

Reply:
let me try delete this` reader` thus the reader inherited from `SerialParsingDriver` is set properly